### PR TITLE
[ISSUE #4794] Fix deserialize ElectMasterResponseHeader attribute value of brokerMemberGroup is null

### LINKED ISSUE #4794: Client deserializes server return ElectMasterResponseHeader, ElectMasterResponseHeader attribute value of brokerMemberGroup is null
**BUG REPORT**

1. Please describe the issue you observed:

when I send CONTROLLER_ELECT_MASTER to controller, then controller process is complete and return RemotingCommand 
image
ElectMasterResponseHeader brokerMemberGroup attribute is not null. but Client get response and ElectMasterResponseHeader,  brokerMemberGroup attribute  is null  after deserialization
image

- What is expected to see?

I'm expected to see brokerMemberGroup is not null

- What did you see instead?

in fact I see ElectMasterResponseHeader attribute brokerMemberGroup is null after deserialization

2. Please tell us about your environment:

branch: develop

3. Other information (e.g. detailed explanation, logs, related issues, suggestions on how to fix, etc):

CommandCustomHeader have custom attribute  like: ElectMasterResponseHeader attribute brokerMemberGroup that instacne of BrokerMemberGroup, Not a Java-defined type like Long, String and so on. the basic type.
Why does this happen? when server need return the response the CommandCustomHeader's attributes will be put into a map, name as map key, and attribute value as map value. the CommandCustomHeader   is not serialized
image

- Question
RemotingCommand CustomHeader is not support custom type serialization? if so why?

### PR DESCRIPTION
**Pull request by mxsm**
**Make sure set the target branch to `develop`**

## What is the purpose of the change

Fix #4794 

## Brief changelog

- Fix deserialize ElectMasterResponseHeader attribute value of brokerMemberGroup is null

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Review by RongtongJin — COMMENTED** [review]
It would be better to remove brokerMemberGroup in ElectMasterResponseHeader.

File: controller/src/main/java/org/apache/rocketmq/controller/impl/manager/ReplicasInfoManager.java [hunk 1]
```diff
@@ -211,8 +211,10 @@ private boolean tryElectMaster(final ControllerResult<ElectMasterResponseHeader>
                 response.setNewMasterAddress(candidate);
                 response.setMasterEpoch(masterEpoch + 1);
                 response.setSyncStateSetEpoch(syncStateSetEpoch);
-                response.setBrokerMemberGroup(buildBrokerMemberGroup(brokerName));
-
+                BrokerMemberGroup brokerMemberGroup = buildBrokerMemberGroup(brokerName);
+                if (null != brokerMemberGroup) {
```
Comment (ShannonDing) [inline controller/src/main/java/org/apache/rocketmq/controller/impl/manager/ReplicasInfoManager.java:215]: what's the default value if buildBrokerMemberGroup is null?

**Comment by hzh0425** [comment]
> It would be better to remove brokerMemberGroup in ElectMasterResponseHeader.

Actually, It's used for notifying brokers, see :
image

**Review by hzh0425 — APPROVED** [review]

**Comment by RongtongJin** [comment]
> > It would be better to remove brokerMemberGroup in ElectMasterResponseHeader.
> 
> Actually, It's used for notifying brokers, see : image

OK, I got it

**Review by RongtongJin — APPROVED** [review]

**Pull request merged**